src/simplemmap.c

```diff
@@ -414,14 +414,46 @@ static LPSTR GetErrorMessage()
     return buf;
 }
 
+#ifdef SIMPLEMMAP
+#define BSIZE 100000
+#include <R_ext/Riconv.h>
+/* Reimplementation of filenameToWchar in main/sysutils.c that converts to
+ * UTF-8 first and then to UTF-16LE (the original version detects the encoding
+ * first and then converts to UCS-2LE) */
+static wchar_t *filenameToWchar(const SEXP fn, const Rboolean expand)
+{
+    static wchar_t filename[BSIZE + 1];
+    const char *inbuf = translateCharUTF8(fn);
+    if (expand) {
+        inbuf = R_ExpandFileNameUTF8(inbuf);
```

Comment:
Compiler complains that it cannot find the declaration of `R_ExpandFileNameUTF8`.

Reply:
The declaration is only available if `Win32` is defined, which appears to be internal. This seems strange for a function that is mentioned in 'Writing R Extensions'.